libopenage/pathfinding/grid.h

```diff
@@ -95,6 +98,12 @@ class Grid {
 	 */
 	void init_portals();
 
+	using portalgraph_t = std::unordered_map<portal_id_t, std::map<portal_id_t, int>>;
```

Comment:
The portal node graph should contain the `PortalNode` instances, not just a mapping between IDs. The creation of `PortalNode` instances is what actually takes the most amount of time.